packages/drupal/silverback_autosave/silverback_autosave.module

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * @file
+ * This module holds autosave form functionality.
+ */
+
+use Drupal\Core\Entity\ContentEntityFormInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\silverback_autosave\Form\AutosaveEntityFormHandler;
+use Drupal\silverback_autosave\Form\AutosaveEntityFormHandlerInterface;
+
+/**
+ * Implements hook_entity_type_alter().
+ *
+ * Sets the default autosave form handler to the entity type if it is not
+ * defined.
+ *
+ * @see \Drupal\Core\Entity\Annotation\EntityType
+ */
+function silverback_autosave_entity_type_alter(array &$entity_types) {
+  // Provide defaults for autosave info.
+  /** @var \Drupal\Core\Entity\EntityTypeInterface[] $entity_types */
+  foreach ($entity_types as $entity_type) {
+    if (!$entity_type->hasHandlerClass('silverback_autosave')) {
+      $entity_type->setHandlerClass('silverback_autosave', AutosaveEntityFormHandler::class);
+    }
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function silverback_autosave_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $active_on = \Drupal::configFactory()->get('silverback_autosave.settings')->get('active_on');
+  $form_object = $form_state->getFormObject();
+
+  if ($form_object instanceof ContentEntityFormInterface) {
+    if (!$active_on['content_entity_forms']) {
+      return;
+    }
+
+    $entity = $form_object->getEntity();
+    $entity_type_id = $entity->getEntityTypeId();
+    $allowed_content_entity_types = \Drupal::configFactory()
+      ->get('silverback_autosave.settings')
+      ->get('allowed_content_entity_types');
+
+    // Autosave is enabled if either no restriction has been made or the current
+    // entity type is part of the restriction and as well either no restriction
+    // on bundle level has been made or the current bundle is part of that
+    // restriction.
+    if (!empty($allowed_content_entity_types)) {
+      if (!isset($allowed_content_entity_types[$entity_type_id]) ||
+        (!empty($allowed_content_entity_types[$entity_type_id]['bundles']) && !isset($allowed_content_entity_types[$entity_type_id]['bundles'][$entity->bundle()]))) {
```

Comment:
This feels like it could be refactored to return early with single if() statements instead of multiple chained together